engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java

```diff
@@ -0,0 +1,1798 @@
+                    ? new LongRollupUniqueOperator(type, resultName, includeNulls, onsAsType, nusAsType)
+                    : new LongChunkedUniqueOperator(type, resultName, includeNulls, exposeInternal, onsAsType,
+                            nusAsType);
+        } else if (type == Short.class || type == short.class) {
+            final short onsAsType =
+                    (onlyNullsSentinel == null) ? NULL_SHORT : ((Number) onlyNullsSentinel).shortValue();
+            final short nusAsType =
+                    (nonUniqueSentinel == null) ? NULL_SHORT : ((Number) nonUniqueSentinel).shortValue();
+            return reaggregated
+                    ? new ShortRollupUniqueOperator(resultName, includeNulls, onsAsType, nusAsType)
+                    : new ShortChunkedUniqueOperator(resultName, includeNulls, exposeInternal, onsAsType, nusAsType);
+        } else {
+            return reaggregated
+                    ? new ObjectRollupUniqueOperator(type, resultName, includeNulls, onlyNullsSentinel,
+                            nonUniqueSentinel)
+                    : new ObjectChunkedUniqueOperator(type, resultName, includeNulls, exposeInternal, onlyNullsSentinel,
+                            nonUniqueSentinel);
+        }
+    }
+
+    private static void checkType(@NotNull final String name, @NotNull final String valueIntent,
+            @NotNull Class<?> expected, final Object value) {
+        expected = getBoxedType(expected);
+        if (value != null && !expected.isAssignableFrom(value.getClass())) {
+            if (isNumeric(expected) && isNumeric(value.getClass())) {
+                if (checkNumericCompatibility((Number) value, expected)) {
+                    return;
+                }
+                throw new IllegalArgumentException(
+                        String.format("For result column %s the %s %s is larger than can be represented with a %s",
+                                name, valueIntent, value, expected.getName()));
+            }
+            throw new IllegalArgumentException(
+                    String.format("For result column %s the %s must be of type %s but is %s",
+                            name, valueIntent, expected.getName(), value.getClass().getName()));
+        }
+    }
+
+    private static boolean checkNumericCompatibility(@NotNull final Number value, @NotNull final Class<?> expected) {
+        if (expected == Byte.class) {
+            return Byte.MIN_VALUE <= value.longValue() && value.longValue() <= Byte.MAX_VALUE;
+        } else if (expected == Short.class) {
+            return Short.MIN_VALUE <= value.longValue() && value.longValue() <= Short.MAX_VALUE;
+        } else if (expected == Integer.class) {
+            return Integer.MIN_VALUE <= value.longValue() && value.longValue() <= Integer.MAX_VALUE;
+        } else if (expected == Long.class) {
+            return new BigInteger(value.toString()).compareTo(BigInteger.valueOf(Long.MIN_VALUE)) >= 0 &&
```

Comment:
Straight up copy-paste from the original code, so this may actually be a question for you.

I assume the question is specifically concerned with narrowing primitive conversion in `Float.longValue()` and `Double.longValue()`?

I suppose we could box the numeric input into a `BigDecimal` if the class of `value` is `Float` or `Double`.

Reply:
I've got a re-write of all of this pushed for your perusal.